malcolm/modules/scanning/controllers/runnablecontroller.py

```diff
@@ -636,18 +712,41 @@ def pause(self, lastGoodStep: ALastGoodStep = -1) -> None:
         self.try_aborting_function(ss.SEEKING, next_state, self.do_pause, lastGoodStep)
 
     def do_pause(self, completed_steps: int) -> None:
+        """Recalculates the number of configured steps
+        Arguments:
+        completed_steps -- Last good step performed
+        """
         self.run_hooks(PauseHook(p, c) for p, c in self.create_part_contexts().items())
-        in_run_steps = completed_steps % self.steps_per_run
-        steps_to_do = self.steps_per_run - in_run_steps
-        part_info = self.run_hooks(
-            ReportStatusHook(p, c) for p, c in self.part_contexts.items()
-        )
-        self.completed_steps.set_value(completed_steps)
-        self.run_hooks(
-            SeekHook(p, c, completed_steps, steps_to_do, part_info, **kwargs)
-            for p, c, kwargs in self._part_params()
-        )
-        self.configured_steps.set_value(completed_steps + steps_to_do)
+
+        if self.use_breakpoints:
+            self.breakpoint_index -= 1
+            # offset = sum(self.steps_per_run[0:self.iBreakpoint])
+            # assert offset < completed_steps
+            in_run_steps = (
+                completed_steps % self.breakpoint_steps[self.breakpoint_index]
+            )
+            steps_to_do = self.breakpoint_steps[self.breakpoint_index] - in_run_steps
```

Comment:
Repeated code between these two branches, can we move it outside the if statement please?